extensions/ql-vscode/src/databases.ts

```diff
@@ -545,6 +560,24 @@ export class DatabaseManager extends DisposableObject {
     return databaseItem;
   }
 
+  /**
+   * Make a database directory available in an extension-controlled location. If the given URI is already
+   * in a controlled location, this method does nothing and returns the argement unchanged. Otherwise, it
+   * copies the given directory to an extension-controlled location and returns the new location.
+   *
+   * @param uri current location of the database directory, which must be a file path pointing to a directory
+   * @returns possibly new location of the database directory, which is guaranteed to be in an extension-controlled location
+   */
+  private async getDatabaseInManagedLocation(uri: vscode.Uri): Promise<vscode.Uri> {
```

Comment:
I prefer avoiding `get*` named methods for things that do side effects. How about something like this:

```suggestion
  private async ensureDatabaseInManagedLocation(uri: vscode.Uri): Promise<vscode.Uri> {
```